src/OpenTelemetry/Logs/BatchLogRecordExportProcessor.cs

```diff
@@ -57,9 +57,14 @@ public override void OnEnd(LogRecord data)
             // happen here.
             Debug.Assert(data != null, "LogRecord was null.");
 
-            data!.BufferLogScopes();
+            data!.Buffer();
```

Comment:
lets add a TODO to add a targeted test for testing https://github.com/open-telemetry/opentelemetry-dotnet/issues/2905

Reply:
I had some tests I did on #2932 so I just updated and added them to this PR.